pkgs/development/ruby-modules/with-packages/default.nix

```diff
@@ -0,0 +1,80 @@
+{ stdenv, lib, buildEnv, buildRubyGem, ruby, gemConfig, makeWrapper }:
+
+/*
+Example usage:
+nix-shell -E "(import <nixpkgs> {}).ruby.withPackages (pkgs: with pkgs; [ pry nokogiri ])"
+
+You can also use this for writing ruby scripts that run anywhere that has nix
+using a nix-shell shebang:
+  #!/usr/bin/env nix-shell
+  #!nix-shell -i ruby -p "ruby.withPackages (pkgs: with pkgs; [ pry nokogiri ])"
+
+
+Run the following in the nixpkgs root directory to update the ruby-packages.nix:
+./maintainers/scripts/update-ruby-packages
+
+To request addition of a gem, just put it into the Gemfile in this directory,
+update ruby-packages.nix, and make a PR mentioning @manveru.
+
+Please note that this will only provide you with libraries you can require
+within ruby or irb. The executables of gems aren't added to your PATH.
+For executables, we recommend
+*/
+
+let
+  functions = import ../bundled-common/functions.nix { inherit lib gemConfig; };
+
+  gems = lib.mapAttrs (name: initialAttrs:
+    let
+      attrs = functions.applyGemConfigs ({ inherit ruby; gemName = name; } // initialAttrs);
+    in
+      buildRubyGem (functions.composeGemAttrs ruby gems name attrs)
+  ) (import ../../../top-level/ruby-packages.nix);
```

Comment:
What is the advantage of creating so much distance between the producer and the consumer of the gems.nix file?

Even the updater script could be moved into this folder so everything is co-located.

Reply:
For the `ruby-packages.nix` I follow the pattern of all the other languages that put them in `top-level`. I agree that it's a bit awkward to have them so separated, but I didn't really want to clutter the `top-level` directory with too much ruby specific stuff.
I can maybe invoke `update-ruby-packages` from the `/maintainers/scripts` and put the actual code in the `with-packages` directory. Again, I tried to be consistent with `update-python-packages` `update-luarocks-packages`.